src/controllers/verify-ketl.ts

```diff
@@ -1,30 +1,97 @@
-import { Body, Controller, Ctx, Post } from 'amala'
+import { Body, Controller, Ctx, Post, Version } from 'amala'
 import { Context } from 'vm'
-import { badRequest } from '@hapi/boom'
+import { badRequest, notFound } from '@hapi/boom'
 import { ethers } from 'ethers'
 import { polygonProvider } from '@/helpers/providers'
 import AttestationType from '@/validators/AttestationType'
+import AttestationTypeList from '@/validators/AttestationTypeList'
 import BalanceUniqueVerifyBody from '@/validators/BalanceUniqueVerifyBody'
 import Email from '@/validators/Email'
 import Signature from '@/validators/Signature'
 import Token from '@/validators/Token'
 import TwitterBody from '@/validators/TwitterBody'
 import VerificationType from '@/models/VerificationType'
 import fetchUserProfile from '@/helpers/twitter/fetchUserProfile'
+import getAllowlistMap from '@/helpers/getAllowlistMap'
 import getBalance from '@/helpers/getBalance'
 import getEmailDomain from '@/helpers/getEmailDomain'
 import hexlifyString from '@/helpers/hexlifyString'
 import sendEmail from '@/helpers/sendEmail'
 import signAttestationMessage from '@/helpers/signatures/signAttestationMessage'
 import zeroAddress from '@/models/zeroAddress'
 
+const allowlistMap = getAllowlistMap()
+
 @Controller('/verify-ketl')
 export default class VerifyKetlController {
+  @Version('0.2.2')
+  @Post('/token')
+  multipleToken(
+    @Ctx() ctx: Context,
+    @Body({ required: true }) { token, types }: AttestationTypeList & Token
+  ) {
+    const attestations = []
+    for (const type of types) {
+      const allowlist = allowlistMap.get(type)
+      if (allowlist?.has(`token:${token}`))
+        attestations.push(signAttestationMessage(type, hexlifyString(token)))
+    }
+
+    if (!attestations.length)
+      return ctx.throw(
+        notFound(
+          `Couldn't find the invitation for this token. Please, try another one`
+        )
+      )
+    return Promise.all(attestations)
```

Comment:
You want to return a promise, that's right, we don't want to resolve it before return?